### ISSUE TITLE
run_analysis_trajectories.py ValueError: supplied range of [inf, inf] is not finite

### ISSUE THREAD

**Issue by heejongkim** [salilab/imp_analysis_tutorial#4]
It's very interesting.
Here's what I did so far.

1. Simply executed run_analysis_trajectories.py with ../modeling/ example (as results are provided in this repo).
this gives no value error or anything but smoothly finish up.
2. performed the sampling with the same data and modeling.py with same 8 5000. Iteration was finished just fine.
When I performed run_analysis_trajectories.py with ../modeling/ smaller (which is the modeling i just executed)
I'm getting the following output with error msg. Does it indicate that there's an issue with modeling?
I tested with conda version and manually compiled version. Both save me more or less consistent result with same type of value error msg.

> $ python ./run_analysis_trajectories.py ../modeling/ smaller
> The mean score, min score, and n frames are:  1 588.325948240789 561.0772547632217 5000
> Trajectory, ts_eqs:  smaller1 [200, 0, 3400, 0]
> The mean score, min score, and n frames are:  6 602.5281745176727 578.3488141088378 5000
> Trajectory, ts_eqs:  smaller6 [0, 200, 0, 0]
> The mean score, min score, and n frames are:  2 632.6239315579007 608.8117325905234 5000
> Trajectory, ts_eqs:  smaller2 [0, 200, 1600, 0]
> The mean score, min score, and n frames are:  8 591.0158963523896 565.5735195176985 5000
> Trajectory, ts_eqs:  smaller8 [600, 600, 2000, 0]
> The mean score, min score, and n frames are:  5 inf inf 5000
> /cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/core/_methods.py:193: RuntimeWarning: invalid value encountered in subtract
>   x = asanyarray(arr - arrmean)
> /home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis/../../lib/PMI_analysis/pyext/src/equilibration.py:101: RuntimeWarning: invalid value encountered in subtract
>   dA_n = A_n.astype(np.float64) - mu_A
> /home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis/../../lib/PMI_analysis/pyext/src/equilibration.py:102: RuntimeWarning: invalid value encountered in subtract
>   dB_n = B_n.astype(np.float64) - mu_B
> Trajectory, ts_eqs:  smaller5 [0, 0, 3800, 0]
> Process Process-2:
> Traceback (most recent call last):
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
>     self.run()
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/multiprocessing/process.py", line 99, in run
>     self._target(*self._args, **self._kwargs)
>   File "/home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis/../../lib/PMI_analysis/pyext/src/analysis_trajectories.py", line 553, in read_traj_info
>     self.plot_scores_restraints(S_tot_scores[['MC_frame']+sel_entries], ts_eq, file_out)
>   File "/home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis/../../lib/PMI_analysis/pyext/src/analysis_trajectories.py", line 627, in plot_scores_restraints
>     n_bins, histtype='step',fill=False, color='orangered',alpha=0.9)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/matplotlib/__init__.py", line 1565, in inner
>     return func(ax, *map(sanitize_sequence, args), **kwargs)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 6649, in hist
>     m, bins = np.histogram(x[i], bins, weights=w[i], **hist_kwargs)
>   File "<__array_function__ internals>", line 6, in histogram
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 795, in histogram
>     bin_edges, uniform_bins = _get_bin_edges(a, bins, range, weights)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 429, in _get_bin_edges
>     first_edge, last_edge = _get_outer_edges(a, range)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 319, in _get_outer_edges
>     "supplied range of [{}, {}] is not finite".format(first_edge, last_edge))
> **ValueError: supplied range of [inf, inf] is not finite**
> All available fields:  ['MC_frame' 'rmf_frame_index' 'Total_Score' 'XLs_Trnka' 'XLs_Chen'
>  'XLs_psi_Trnka' 'XLs_psi_Chen' 'CR_Rpb2' 'CR_Rpb11' 'CR_Rpb3' 'CR_Rpb9'
>  'CR_Rpb10' 'CR_Rpb6' 'CR_Rpb12' 'CR_Rpb8' 'CR_Rpb1' 'CR_Rpb4' 'CR_Rpb7'
>  'CR_Rpb5' 'EV_sum' 'GaussianEMRestraint_None'
>  'GaussianEMRestraint_None_CCC' 'GaussianEMRestraint_sigma_None' 'traj'
>  'rmf3_file' 'XLs_sum' 'CR_sum' 'half']
> Number of unique clusters:  3 [-1  0  1]
> Generating HDBSCAN clustering plot ...
> Selecting and writing models to extract ...
> Clustering summary: 
>          Total_Score      EV_sum     XLs_sum  GaussianEMRestraint_None  N_models  N_A   N_B
> cluster                                                                                    
>  1        593.346315  300.079624  144.830518                145.810335      5823  527  5296
>  0        626.219631  328.401122  145.844188                148.907098      2032    6  2026
> -1        636.248459  326.632034  149.667570                152.847302       945   67   878
> Summarize XLs, unique_clusters [-1  0  1]

If you have any insights what might've caused the issue and prevent this, I would really appreciate your help.

Thank you so much.

best,
hee jong

**Comment by ichem001** [comment]
Hi Hee Jong, 
Did you check the `rmf` files to make sure the trajectories behaves nicely.
Specifically, the line `[t]he mean score, min score, and n frames are: 5 inf inf 5000` tells me something went awry in scoring. That's probably why you see a bunch of errors. 

If you run the `modeling.py` provided as is with the latest release of IMP, do you run in the same issue?

Hope that helps.

**Comment by heejongkim** [comment]
Hi,

Yes. I was worried about possible issue from software installation.
I tried with conda version and clone this repo and ran the bash script with N=8 and num_frame 5000.
No modeling.py or anything was edited.

> ---Omitted---
> --- writing coordinates
> --- frame 4983 score 575.0482442634341 
> --- writing coordinates
> --- frame 4996 score 578.0550992361991 
> --- writing coordinates
> --- frame 4984 score 578.3017777219093 
> --- writing coordinates
> --- frame 4997 score 579.7311480214746 
> --- writing coordinates
> --- frame 4985 score 571.2441985368224 
> --- writing coordinates
> --- frame 4998 score 581.3224214175532 
> --- writing coordinates
> --- frame 4986 score 571.2199844997479 
> --- writing coordinates
> --- frame 4999 score 582.0431884004297 
> --- writing coordinates
> closing production rmf files
> --- frame 4987 score 567.3685279657111 
> --- writing coordinates
> --- frame 4988 score 572.277570421417 
> --- writing coordinates
> --- frame 4989 score 572.9263003656733 
> --- writing coordinates
> --- frame 4990 score 575.2213828017673 
> --- writing coordinates
> --- frame 4991 score 574.3569486547268 
> --- writing coordinates
> --- frame 4992 score 569.2967355978844 
> --- writing coordinates
> --- frame 4993 score 567.5565484518347 
> --- writing coordinates
> --- frame 4994 score 569.9603991249345 
> --- writing coordinates
> --- frame 4995 score 575.3952856828439 
> --- writing coordinates
> --- frame 4996 score 577.4914595708017 
> --- writing coordinates
> --- frame 4997 score 576.6998593557105 
> --- writing coordinates
> --- frame 4998 score 574.2655274080723 
> --- writing coordinates
> --- frame 4999 score 577.6530254130271 
> --- writing coordinates
> closing production rmf files

after it's done, i went to analysis folder and executed ./run_analysis_trajectories.py as instructed in README and got the error above.

Please let me know if any other information might've been useful to narrow down the issue.
Also, I will take a look at rmf files in the meantime.

Thanks.

**Comment by ichem001** [comment]
Yes - look at the trajectories. Can you also extract the Total_Score from your output files to see if one or more of them got a `inf` score. That `inf` score will give you the frame where sampling broke and put you in a better position to fix the issue.

**Comment by heejongkim** [comment]
Thanks.
I'm downloading those rmf files from server now.
Do you have any preference or recommendation for how to look at that data? grep? or any imp/pmi tools to extract them out?
Thank you so much!

**Comment by heejongkim** [comment]
This is from plot_scores_1.pdf that came from the execution.
![Screen Shot 2021-01-06 at 3 37 06 AM](https://user-images.githubusercontent.com/10135880/103747386-9108d880-4fd0-11eb-9498-10f589a98de8.png)

**Comment by tanmoy7989** [comment]
8 runs with 5000 frames (total 40000)  might be a case of undersampling. No wonder you have zero equilibrated frames (as can also be seen from the position of the gray line in the upper panel here), and so the rest of PMI_Analysis barfs. The README recommends 50000 frames for each run, so you are off by an order of magnitude.

**Comment by heejongkim** [comment]
Interesting.
This repo provided example result with 8runs with 5,000 frames each if i understood correctly.
So, the example results are truncated from 50,000 frames for each run for the sake of github size limit?

I actually did the 48 runs with 50,000 frames each and running analysis again as of now. As soon as i get the final verdict, i will be back.

Thank you.

**Comment by tanmoy7989** [comment]
You are correct, the repo does provide 8 runs each with 5000 frames. But not sure, why the equilibration step gets messed up in your case. Almost certainly, something weird has happened for your 8th run (i.e. smaller8), which causes the infinity at the equlibration step and all the subsequent errors thereafter. Try and delete that folder (i.e. smaller8) and re-run the analysis.

**Comment by heejongkim** [comment]
Thank you for the confirmation.
Here's after removing the folder, "smaller8"

> $ python ./run_analysis_trajectories.py ../modeling/ smaller
> The mean score, min score, and n frames are:  1 588.325948240789 561.0772547632217 5000
> Trajectory, ts_eqs:  smaller1 [200, 0, 3400, 0]
> The mean score, min score, and n frames are:  6 602.5281745176727 578.3488141088378 5000
> Trajectory, ts_eqs:  smaller6 [0, 200, 0, 0]
> The mean score, min score, and n frames are:  2 632.6239315579007 608.8117325905234 5000
> Trajectory, ts_eqs:  smaller2 [0, 200, 1600, 0]
> The mean score, min score, and n frames are:  5 inf inf 5000
> /cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/core/_methods.py:193: RuntimeWarning: invalid value encountered in subtract
>   x = asanyarray(arr - arrmean)
> /home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis_smaller/../../lib/PMI_analysis/pyext/src/equilibration.py:101: RuntimeWarning: invalid value encountered in subtract
>   dA_n = A_n.astype(np.float64) - mu_A
> /home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis_smaller/../../lib/PMI_analysis/pyext/src/equilibration.py:102: RuntimeWarning: invalid value encountered in subtract
>   dB_n = B_n.astype(np.float64) - mu_B
> Trajectory, ts_eqs:  smaller5 [0, 0, 3800, 0]
> Process Process-2:
> Traceback (most recent call last):
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
>     self.run()
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/multiprocessing/process.py", line 99, in run
>     self._target(*self._args, **self._kwargs)
>   File "/home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis_smaller/../../lib/PMI_analysis/pyext/src/analysis_trajectories.py", line 553, in read_traj_info
>     self.plot_scores_restraints(S_tot_scores[['MC_frame']+sel_entries], ts_eq, file_out)
>   File "/home/kim/conda_imp/original/imp_analysis_tutorial/rnapolii/analysis_smaller/../../lib/PMI_analysis/pyext/src/analysis_trajectories.py", line 627, in plot_scores_restraints
>     n_bins, histtype='step',fill=False, color='orangered',alpha=0.9)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/matplotlib/__init__.py", line 1565, in inner
>     return func(ax, *map(sanitize_sequence, args), **kwargs)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 6649, in hist
>     m, bins = np.histogram(x[i], bins, weights=w[i], **hist_kwargs)
>   File "<__array_function__ internals>", line 6, in histogram
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 795, in histogram
>     bin_edges, uniform_bins = _get_bin_edges(a, bins, range, weights)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 429, in _get_bin_edges
>     first_edge, last_edge = _get_outer_edges(a, range)
>   File "/cm/shared/apps/python-3.7.7/lib/python3.7/site-packages/numpy/lib/histograms.py", line 319, in _get_outer_edges
>     "supplied range of [{}, {}] is not finite".format(first_edge, last_edge))
> ValueError: supplied range of [inf, inf] is not finite
> All available fields:  ['MC_frame' 'rmf_frame_index' 'Total_Score' 'XLs_Trnka' 'XLs_Chen'
>  'XLs_psi_Trnka' 'XLs_psi_Chen' 'CR_Rpb2' 'CR_Rpb11' 'CR_Rpb3' 'CR_Rpb9'
>  'CR_Rpb10' 'CR_Rpb6' 'CR_Rpb12' 'CR_Rpb8' 'CR_Rpb1' 'CR_Rpb4' 'CR_Rpb7'
>  'CR_Rpb5' 'EV_sum' 'GaussianEMRestraint_None'
>  'GaussianEMRestraint_None_CCC' 'GaussianEMRestraint_sigma_None' 'traj'
>  'rmf3_file' 'XLs_sum' 'CR_sum' 'half']
> Number of unique clusters:  4 [-1  0  1  2]
> Generating HDBSCAN clustering plot ...
> Selecting and writing models to extract ...
> Clustering summary: 
>          Total_Score      EV_sum     XLs_sum  GaussianEMRestraint_None  N_models  N_A   N_B
> cluster                                                                                    
>  1        574.863362  299.633737  136.033941                137.256990       326  324     2
>  2        598.318535  299.326789  148.147084                148.461681      3216   32  3184
> -1        626.481015  319.428779  149.011927                151.868456      1203  238   965
>  0        626.637850  328.602988  145.919631                149.001976      2055    6  2049
> Summarize XLs, unique_clusters [-1  0  1  2]

at least I do get a lot more output files than before.

**Comment by tanmoy7989** [comment]
So basically you need to remove whichever run is causing the statement
```The mean score, min score, and n frames are: 5 inf inf 5000```

Sorry, I hurriedly wrote smaller8, but looks like it some other run. So here are a few things to try, **in this order**:
a) Delete one folder at a time from smaller1 to smaller8 (keep backups obv) and see when the above message goes away.
b) Once you've isolated that folder, as @ichem001  pointed out earlier, check the rmf files for that run to see everything is nice. 
c) Re-run against the most recent stable IMP version.
d) See if all problems go away when you sample more (e.g. 50,000 frames)

**Comment by heejongkim** [comment]
Thank you for your guidance.
I will follow them sequentially and get back to you.
Thank you.

**Comment by heejongkim** [comment]
a) narrowed down to "smaller5" result was the causing factor. By removing that folder, i got the following output:

> $ python ./run_analysis_trajectories.py ../modeling/ smaller
> The mean score, min score, and n frames are:  1 588.325948240789 561.0772547632217 5000
> Trajectory, ts_eqs:  smaller1 [200, 0, 3400, 0]
> The mean score, min score, and n frames are:  6 602.5281745176727 578.3488141088378 5000
> Trajectory, ts_eqs:  smaller6 [0, 200, 0, 0]
> The mean score, min score, and n frames are:  2 632.6239315579007 608.8117325905234 5000
> Trajectory, ts_eqs:  smaller2 [0, 200, 1600, 0]
> The mean score, min score, and n frames are:  8 591.0158963523896 565.5735195176985 5000
> Trajectory, ts_eqs:  smaller8 [600, 600, 2000, 0]
> The mean score, min score, and n frames are:  7 647.3353137464535 616.6781799610642 5000
> Trajectory, ts_eqs:  smaller7 [1400, 1800, 0, 0]
> The mean score, min score, and n frames are:  4 598.6215376040226 570.7048313292796 5000
> Trajectory, ts_eqs:  smaller4 [1200, 0, 3600, 0]
> The mean score, min score, and n frames are:  3 605.8135490517146 575.126562691534 5000
> Trajectory, ts_eqs:  smaller3 [0, 3200, 2200, 0]
> All available fields:  ['MC_frame' 'rmf_frame_index' 'Total_Score' 'XLs_Trnka' 'XLs_Chen'
>  'XLs_psi_Trnka' 'XLs_psi_Chen' 'CR_Rpb2' 'CR_Rpb11' 'CR_Rpb3' 'CR_Rpb9'
>  'CR_Rpb10' 'CR_Rpb6' 'CR_Rpb12' 'CR_Rpb8' 'CR_Rpb1' 'CR_Rpb4' 'CR_Rpb7'
>  'CR_Rpb5' 'EV_sum' 'GaussianEMRestraint_None'
>  'GaussianEMRestraint_None_CCC' 'GaussianEMRestraint_sigma_None' 'traj'
>  'rmf3_file' 'XLs_sum' 'CR_sum' 'half']
> Number of unique clusters:  3 [-1  0  1]
> Generating HDBSCAN clustering plot ...
> Selecting and writing models to extract ...
> Clustering summary:
>          Total_Score      EV_sum     XLs_sum  GaussianEMRestraint_None  N_models   N_A   N_B
> cluster
>  0        593.641111  300.457167  144.835607                145.591608      7187  6802   385
>  1        631.215529  334.956369  146.668423                146.033119      4268    99  4169
> -1        646.136453  335.997872  150.404484                152.336440       745   299   446
> Summarize XLs, unique_clusters [-1  0  1]

no error msg anymore

b) i checked the smaller5's rmf file and as you can see the screenshot, it went wild haha.
![Screen Shot 2021-01-06 at 4 57 40 PM](https://user-images.githubusercontent.com/10135880/103823596-635b7800-5040-11eb-9004-b59cc0723146.png)


c). this is already done by 2.14.0 from both Conda and Source code compiled. It was at least consistent to emit the same error.

d)
Here's 48 50,000 output:

> The mean score, min score, and n frames are:  23 594.6378097432608 559.5008269736215 50000
> Trajectory, ts_eqs:  results23 [26400, 4800, 15400, 0]
> The mean score, min score, and n frames are:  29 598.6009184737547 571.0833453252288 50000
> Trajectory, ts_eqs:  results29 [0, 0, 45400, 0]
> The mean score, min score, and n frames are:  11 588.2188771945868 559.4894668586915 50000
> Trajectory, ts_eqs:  results11 [1600, 2600, 44600, 0]
> The mean score, min score, and n frames are:  47 599.6856829028568 574.4630490783302 50000
> Trajectory, ts_eqs:  results47 [1800, 0, 28600, 0]
> The mean score, min score, and n frames are:  30 592.5071731178075 559.4397779365083 50000
> Trajectory, ts_eqs:  results30 [21400, 2000, 36200, 0]
> The mean score, min score, and n frames are:  38 598.8113930377021 572.0510587455051 50000
> Trajectory, ts_eqs:  results38 [1400, 1400, 9000, 0]
> The mean score, min score, and n frames are:  45 592.8791102912219 561.4188444227931 50000
> Trajectory, ts_eqs:  results45 [13800, 0, 40600, 0]
> The mean score, min score, and n frames are:  32 589.4937188593461 563.2881299650023 50000
> Trajectory, ts_eqs:  results32 [1400, 0, 10200, 0]
> The mean score, min score, and n frames are:  21 603.0624421284505 573.7698043907619 50000
> Trajectory, ts_eqs:  results21 [4000, 2600, 8200, 0]
> The mean score, min score, and n frames are:  8 638.5888781431414 602.0227179191074 50000
> Trajectory, ts_eqs:  results8 [8800, 9200, 11000, 0]
> The mean score, min score, and n frames are:  2 593.2728092096912 561.50465144032 50000
> Trajectory, ts_eqs:  results2 [42000, 0, 43200, 0]
> The mean score, min score, and n frames are:  13 643.3265758673274 612.325756172143 50000
> Trajectory, ts_eqs:  results13 [0, 0, 17800, 0]
> The mean score, min score, and n frames are:  19 621.819887954323 588.358054801404 50000
> Trajectory, ts_eqs:  results19 [800, 800, 27600, 200]
> The mean score, min score, and n frames are:  14 589.5677890569399 560.9985197892184 50000
> Trajectory, ts_eqs:  results14 [200, 0, 44800, 0]
> The mean score, min score, and n frames are:  26 598.0419715820877 573.9857847421684 50000
> Trajectory, ts_eqs:  results26 [0, 0, 30200, 0]
> The mean score, min score, and n frames are:  5 594.073796666932 559.2117577919769 50000
> Trajectory, ts_eqs:  results5 [4400, 4400, 35600, 3200]
> The mean score, min score, and n frames are:  48 595.9735680289059 562.7697874469095 50000
> Trajectory, ts_eqs:  results48 [30400, 0, 45400, 0]
> The mean score, min score, and n frames are:  35 588.0790913948318 560.7112722774674 50000
> Trajectory, ts_eqs:  results35 [800, 800, 0, 0]
> The mean score, min score, and n frames are:  42 595.0886369548676 561.5786279679955 50000
> Trajectory, ts_eqs:  results42 [200, 200, 0, 0]
> The mean score, min score, and n frames are:  31 601.2469539518144 565.0576294136945 50000
> Trajectory, ts_eqs:  results31 [1400, 2400, 1800, 0]
> The mean score, min score, and n frames are:  46 592.4313376697847 562.7551801807408 50000
> Trajectory, ts_eqs:  results46 [9000, 0, 47800, 0]
> The mean score, min score, and n frames are:  28 599.5121368589284 561.3757324824433 50000
> Trajectory, ts_eqs:  results28 [14400, 5200, 20800, 0]
> The mean score, min score, and n frames are:  1 602.7403482382938 564.4810400613555 50000
> Trajectory, ts_eqs:  results1 [1600, 1600, 45800, 600]
> The mean score, min score, and n frames are:  22 609.3219093862726 572.2630611694366 50000
> Trajectory, ts_eqs:  results22 [9000, 9000, 400, 0]
> The mean score, min score, and n frames are:  10 597.9824996855476 574.2005818785531 50000
> Trajectory, ts_eqs:  results10 [0, 0, 0, 0]
> The mean score, min score, and n frames are:  17 597.4104270619117 564.299358413238 50000
> Trajectory, ts_eqs:  results17 [400, 600, 41600, 0]
> The mean score, min score, and n frames are:  6 590.6031048440748 562.8166973756523 50000
> Trajectory, ts_eqs:  results6 [13000, 200, 12800, 0]
> The mean score, min score, and n frames are:  25 590.3002664015922 563.3040137659247 50000
> Trajectory, ts_eqs:  results25 [800, 800, 25600, 0]
> The mean score, min score, and n frames are:  41 598.8735241133261 574.1914854460015 50000
> Trajectory, ts_eqs:  results41 [1400, 0, 26200, 0]
> The mean score, min score, and n frames are:  36 589.4369276259632 557.7936915321364 50000
> Trajectory, ts_eqs:  results36 [42200, 0, 45800, 0]
> All available fields:  ['MC_frame' 'rmf_frame_index' 'Total_Score' 'XLs_Trnka' 'XLs_Chen'
>  'XLs_psi_Trnka' 'XLs_psi_Chen' 'CR_Rpb10' 'CR_Rpb1' 'CR_Rpb6' 'CR_Rpb9'
>  'CR_Rpb12' 'CR_Rpb7' 'CR_Rpb2' 'CR_Rpb3' 'CR_Rpb8' 'CR_Rpb11' 'CR_Rpb5'
>  'CR_Rpb4' 'EV_sum' 'GaussianEMRestraint_None'
>  'GaussianEMRestraint_None_CCC' 'GaussianEMRestraint_sigma_None' 'traj'
>  'rmf3_file' 'XLs_sum' 'CR_sum' 'half']
> Number of unique clusters:  4 [-1  0  1  2]
> Generating HDBSCAN clustering plot ...
> Selecting and writing models to extract ...
> Clustering summary:
>          Total_Score      EV_sum     XLs_sum  GaussianEMRestraint_None  N_models     N_A     N_B
> cluster
>  0        598.240209  305.068476  145.622026                144.443168    912400  528000  384400
>  1        721.754602  315.185305  149.664048                254.613820      2224       0    2224
>  2        755.096483  316.677329  151.101816                284.269830     30054       0   30054
> -1        776.261876  331.843353  156.436978                281.312505       922       0     922
> Summarize XLs, unique_clusters [-1  0  1  2]

i don't see that value error msg anymore! I think sampling amount helped

In conclusion, it seems like having enough sampling resolve the issue but why provided result was fine with that little amount of samplings while my attempt to reproduce the result wasn't ok remain unclear.

**Comment by tanmoy7989** [comment]
Well, sampling is stochastic. So, your run # 5 may not have explored enough compared to the runs in this repository. Perhaps your randomized initial state was such that it got trapped in a local minima. To be sure, you could look at the stat file from the failed run #5. But usually the easiest thing to do is increase sampling, which you did.
